libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -875,13 +875,20 @@ Vector3f AC_AttitudeControl::euler_accel_limit(const Vector3f &euler_rad, const
     return rot_accel;
 }
 
-// Shifts earth frame yaw target by yaw_shift_cd. yaw_shift_cd should be in centidegrees and is added to the current target heading
-void AC_AttitudeControl::shift_ef_yaw_target(float yaw_shift_cd)
+// Sets yaw target to vehicle heading and sets yaw rate to zero
+void AC_AttitudeControl::set_yaw_target_to_current_heading()
 {
-    float yaw_shift = radians(yaw_shift_cd * 0.01f);
+    // move attitude target to current heading
+    float yaw_shift = _ahrs.yaw - _euler_angle_target.z;
     Quaternion _attitude_target_update;
     _attitude_target_update.from_axis_angle(Vector3f{0.0f, 0.0f, yaw_shift});
```

Comment:
It is existing code but in any case, we have noticed in other situations where we are using "xxx_axis_angle" that the result is not correct (see PR https://github.com/ArduPilot/ardupilot/pull/17551).  I think we should check if the resulting target does exactly match the ahrs.yaw.

Reply:
So, a bit as expected the values are not exactly the same especially when the vehicle is leaning a lot or pointing straight up.  When horizontal they are within a degree of each other.
![image](https://user-images.githubusercontent.com/1498098/119607252-ee41e100-be2e-11eb-8bf5-a6bf3997a7c7.png)